### ISSUE TITLE
Syntax error with input redirection combined with process substitution

### ISSUE THREAD

**Issue by 39555** [reubeno/brush#144]
Hi! Very cool project! I started using brush as my multi platform bash and appreciate your work! I want to document a syntax error when input redirection is combined with process substitution:
1.
```brush
$ cat < <(echo hello)
ERROR brush_core::shell: main: syntax error near token `<' (line 1 col 7)
```

2. or
```bash
while IFS= read -r line; do
        echo $line
# do something with the file content before read lines
done < <(tac "test.txt" | awk '{$1=$1};1' | grep -v '^#')
```

**Comment by reubeno** [comment]
Thanks for the feedback! I'm glad to hear you're finding it useful and appreciate getting reports on bugs like these. I can reproduce the issue you describe very easily--thanks for the simple steps.

I'll do my best to look into this as soon as I can. (Although you're not expected to do so, if you happen to be interested to look into it / contributing yourself, you're welcome to do so and I'd be happy to provide some pointers to get started.)

**Comment by reubeno** [comment]
I looked into this and realized the code base didn't have proper support for using a process substitution like the above (e.g., `< <(...)`, `> >(...)`) anywhere a redirection could be done to/from a filename. Given that, the proper fix is a bit more involved so I figured I'd take a crack at it.

I've posted a PR with a fix for these specific cases and a few others I spotted along the way.

With the fix in place, I confirmed both of the example cases posted above seem to work as expected.
